src/libs/tryResolveUrlFromApiRoot.ts

```diff
@@ -19,6 +19,7 @@ const ORIGIN_PATTERN = new RegExp(`^(${ORIGINS_TO_REPLACE.join('|')})`);
  */
 function tryResolveUrlFromApiRoot(url: string): string;
 function tryResolveUrlFromApiRoot(url: number): number;
+function tryResolveUrlFromApiRoot(url: string | number): string | number;
```

Comment:
This shouldn't be necessary 🤔 This is already defined under:
```ts
function tryResolveUrlFromApiRoot(url: string | number): string | number {
```

Reply:
Do you have a better solution to 
`The call would have succeeded against this implementation, but implementation signatures of overloads are not externally visible.` 
remove it and you will see it in `ReportActionItemImage.tsx` looks like implementation signatures are not visible externally

Reply:
I think this is our best bet:
```ts
/**
 * When possible this function resolve URLs relative to API ROOT
 * - Absolute URLs like `/{path}`, become: `https://{API_ROOT}/{path}`
 * - Similarly for prod or staging URLs we replace the `https://www.expensify`
 * or `https://staging.expensify` part, with `https://{API_ROOT}`
 * - Unmatched URLs (non expensify) are returned with no modifications
 */
function tryResolveUrlFromApiRoot(url: string | number): string | number {
    // in native, when we import an image asset, it will have a number representation which can be used in `source` of Image
    // in this case we can skip the url resolving
    if (typeof url === 'number') {
        return url;
    }
    const apiRoot = ApiUtils.getApiRoot({shouldUseSecure: false} as Request);
    return url.replace(ORIGIN_PATTERN, apiRoot);
}
```

Reply:
You suggesting removing the overload signatures?? I mean there is an advantage to the overloads as it will enforce `number` return when the argument type is `number` and same for `string` too

Reply:
What is the point of overloading if you need to add this? 😅 
```ts
function tryResolveUrlFromApiRoot(url: string | number): string | number;
```

Reply:
When the parameter is `number` we will know that the return will be `number` (same for `string`) but when it is `string | number` we will have `string | number` return type that's the point! 👍

Reply:
Thank you for explanation, I thought it worked a bit differently 👍